actionpack/test/dispatch/request/session_test.rb

```diff
@@ -103,6 +103,19 @@ def test_fetch
         end
       end
 
+      def test_with_indifferent_access
+        env = {}
+        s = Session.create(store, env, {})
+
+        s[:one] = { test: "deep" }
+        s[:two] = { "test" => "deep" }
+
+        s2 = Session.create(store, env, {})
```

Comment:
@repinel Very weird, when I was making it, I was sure I tried it. What version of Ruby are you using?